ocaml/typing/printtyp.ml

```diff
@@ -1435,7 +1435,9 @@ let rec tree_of_typexp mode alloc_mode ty =
         let arg_mode = Alloc.zap_to_legacy marg in
         let t1 =
           if is_optional l then
-            match get_desc (tpoly_get_mono ty1) with
+            match
+              get_desc (Ctype.expand_head !printing_env (tpoly_get_mono ty1))
```

Comment:
should we keep the original code as quick-path?

Reply:
I think you probably don't need it. In the common case this is basically just a single look up for the `option` path in the environment.